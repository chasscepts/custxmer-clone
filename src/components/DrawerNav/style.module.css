.drawerNavContainer {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100vh;
  display: none;
  flex-direction: column;
  overflow: hidden;
  background-color: #eee;
  padding: 20px;
  z-index: 10000;
}

.drawerHeader {
  padding: 50px 0 30px;
  display: flex;
  justify-content: flex-end;
}

.drawerCloseBtn {
  color: #333;
  font-size: 22px;
  cursor: pointer;
}

.listItem {
  display: flex;
  justify-content: center;
  padding: 10px;
}

.link {
  border-bottom: 2px solid transparent;
  color: #333;
  font-size: 16px;
  transition: all 0.3s ease;
}

.link:hover {
  color: #666;
}

.link.active {
  border-bottom-color: #333;
}

.link.active:hover {
  border-bottom-color: #666;
}

@media only screen and (max-width: 768px) {
  .drawerNavContainer.open {
    display: flex;
  }
}
