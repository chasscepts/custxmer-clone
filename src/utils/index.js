export const paths = {
  HOME: '/',
  ABOUT: '/about',
  SERVICES: '/services',
  CONTACT: '/contact',
};

/**
 * Format path to application specification
 * @param {string} path windows path to fromat
 */
export const formatPath = (path) => {
  if (path === paths.HOME) {
    return '/Home';
  }

  const parts = path.split('/').filter((p) => p);

  return `/${parts.map((p) => `${p[0].toUpperCase()}${p.substring(1).toLowerCase()}`).join('/')}`;
};
