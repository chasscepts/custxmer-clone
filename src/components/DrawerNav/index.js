import { Link, useLocation } from 'react-router-dom';
import PropTypes from 'prop-types';
import { paths } from '../../utils';
import css from './style.module.css';

const DrawerNav = ({ isOpen, onClose }) => {
  const { pathname } = useLocation();

  return (
    <div className={`${css.drawerNavContainer} ${isOpen ? css.open : ''}`}>
      <div className={css.drawerHeader}>
        <button className={css.drawerCloseBtn} onClick={onClose}>
          <i className="fa-solid fa-xmark" />
        </button>
      </div>
      <nav>
        <ul className="plain-list">
          {Object.keys(paths).map((key) => (
            <li key={key} className={css.listItem}>
              <Link
                className={`${css.link} ${pathname === paths[key] ? css.active : ''}`}
                to={paths[key]}
                onClick={onClose}
              >
                {key}
              </Link>
            </li>
          ))}
        </ul>
      </nav>
    </div>
  );
};

DrawerNav.propTypes = {
  isOpen: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
};

DrawerNav.defaultProps = {
  isOpen: false,
};

export default DrawerNav;
