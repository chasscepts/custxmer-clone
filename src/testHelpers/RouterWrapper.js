import { MemoryRouter } from "react-router";
import PropTypes from 'prop-types';
import { paths } from "../utils";

const RouterWrapper = ({ children }) => (
  <MemoryRouter initialEntries={Object.values(paths)} initialIndex={0}>
    {children}
  </MemoryRouter>
);

RouterWrapper.propTypes = {
  children: PropTypes.node,
};

RouterWrapper.defaultProps = {
  children: null,
};

export default RouterWrapper;
