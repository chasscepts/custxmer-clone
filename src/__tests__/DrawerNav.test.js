import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import RouterWrapper from '../testHelpers/RouterWrapper';
import DrawerNav from '../components/DrawerNav';
import { paths } from '../utils';

describe('Drawer Navigation', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <RouterWrapper>
        <DrawerNav onClose={() => {}} />
      </RouterWrapper>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
    tree.unmount();
  });

  it('displays all the links', () => {
    render(
      <RouterWrapper>
        <DrawerNav onClose={() => {}} />
      </RouterWrapper>,
    );

    Object.keys(paths).forEach((key) => expect(screen.getByText(key)).toBeInTheDocument());
  });
});
