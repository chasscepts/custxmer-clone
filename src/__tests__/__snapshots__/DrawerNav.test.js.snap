// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Navigation matches snapshot 1`] = `
<div
  className="drawerNavContainer "
>
  <div
    className="drawerHeader"
  >
    <button
      className="drawerCloseBtn"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa-solid fa-xmark"
      />
    </button>
  </div>
  <nav>
    <ul
      className="plain-list"
    >
      <li
        className="listItem"
      >
        <a
          className="link active"
          href="/"
          onClick={[Function]}
        >
          HOME
        </a>
      </li>
      <li
        className="listItem"
      >
        <a
          className="link "
          href="/about"
          onClick={[Function]}
        >
          ABOUT
        </a>
      </li>
      <li
        className="listItem"
      >
        <a
          className="link "
          href="/services"
          onClick={[Function]}
        >
          SERVICES
        </a>
      </li>
      <li
        className="listItem"
      >
        <a
          className="link "
          href="/contact"
          onClick={[Function]}
        >
          CONTACT
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
